srtcore/group.cpp

```diff
@@ -1918,6 +1918,218 @@ struct FLookupSocketWithEvent_LOCKED
     }
 };
 
+vector<CUDTSocket*> CUDTGroup::recv_CollectNonBroken(set<CUDTSocket*>& broken)
```

Comment:
As you use already one fill-by-ref argument and nontrivial type, maybe it would be better to return void and fill both by reference. Also a `bool` return to sign that there are no available links at all can be considered, if it makes sense.